### ISSUE TITLE
serverless offline sometimes crashes when editing Elm files

### ISSUE THREAD

**Issue by MartinSStewart** [the-sett/elm-serverless#5]
Sometimes when I make changes to Elm files and hit save, I get the normal build output
```
Started compiling Elm..
Time: 805ms
                  Asset     Size  Chunks                    Chunk Names
   src/Analytics/api.js  5.28 MB       0             [big]  src/Analytics/api
src/SwitchDevice/api.js   537 kB       1  [emitted]  [big]  src/SwitchDevice/api
   [9] ./src-bridge/logger.js 26 bytes {0} {1}
  [21] ../node_modules/uuid/index.js 120 bytes {0} {1}
  [33] ./src-bridge/index.js 1.36 kB {0} {1}
  [35] ./src-bridge/pool.js 714 bytes {0} {1}
  [36] ./src-bridge/request-handler.js 1.44 kB {0} {1}
  [56] ../node_modules/uuid/v1.js 3.34 kB {0} {1}
  [57] ../node_modules/uuid/v4.js 680 bytes {0} {1}
  [59] ./src-bridge/response-handler.js 1.38 kB {0} {1}
  [60] ./src-bridge/validate.js 449 bytes {0} {1}
 [130] ./src/Analytics/api.js 1.25 kB {0}
 [131] ./src/Analytics/API.elm 128 kB {0}
 [132] ../node_modules/aws-sdk/lib/aws.js 159 bytes {0}
 [240] ../node_modules/aws-sdk/clients/all.js 7.82 kB {0}
 [902] ./src/SwitchDevice/api.js 240 bytes {1}
 [903] ./src/SwitchDevice/API.elm 174 kB {1} [built]
    + 889 hidden modules
```
followed immediately by this
```
Assertion failed: (handle->flags & UV_HANDLE_CLOSING), function uv__finish_close, file ../deps/uv/src/unix/core.c, line 246.
sh: line 1:  2679 Abort trap: 6           ( serverless offline --port 3001 )
```

This is on MacOS, using Elm 0.19.1. Unfortunately this doesn't happen consistently and I haven't been able to figure out a pattern to it.

**Comment by rupertlssmith** [comment]
I think this might be in the serverless process itself?

`Abort trap: 6           ( serverless offline --port 3001 )`

But of course that is ultimately wrapping the Elm code, and the JS bridge code, so its not impossible the error could be originating there.

Was there any more stack trace than the 2 lines above?

There is a script under `scripts/test-server.js` that I used to start and stop serverless when running th tests or examples. This uses `process.kill(server.pid);` to kill off the old serverless instance when restarting. It seems to work find on my Debian.

Unfortunately I don't have a Mac.

**Comment by rupertlssmith** [comment]
> Sometimes when I make changes to Elm files and hit save, I get the normal build output

How are you running this build loop?

**Comment by MartinSStewart** [comment]
> Was there any more stack trace that the 2 lines above?

No, before that was just the normal build output that I posted.

> There is a script under scripts/test-server.js that I used to start and stop serverless when running th tests or examples.

I removed that script and instead turn off serverless offline by pressing Ctrl + C in the terminal. I guess it's possible that Ctrl + C isn't turning off the server but if that were the case then I think serverless offline would warn me that the port is already in use when I try starting it again.

> How are you running this build loop?

I run the following `serverless offline --port 3001`. [Here](https://github.com/the-sett/elm-serverless/files/3847192/config.zip) are the serverless.yml and webpack.config.js files im using. They are based off the demo examples.
